framework/src/problems/FEProblemBase.C

```diff
@@ -266,6 +266,14 @@ FEProblemBase::validParams()
       false,
       "Set to true to allow convergence even though the solution has been marked as 'invalid'");
 
+  params.addParam<bool>("immediately_print_invalid_solution",
+                        false,
+                        "Whether or not to report invalid solution warnings at the time the "
+                        "warning is produced instead of after the calculation");
```

Comment:
do we need this? who asked for this?
@friedmud @loganharbour ?

Reply:
Derek wants to have this capability

Reply:
ok I can go either way. It s on the verbose side of things, but if it allows better debugging, why not